public/app/features/explore/ExplorePage.tsx

```diff
@@ -32,10 +31,9 @@ const styles = {
 };
 
 export default function ExplorePage(props: GrafanaRouteComponentProps<{}, ExploreQueryParams>) {
-  useStopQueries();
   useTimeSrvFix();
   useStateSync(props.queryParams);
-  useExplorePageTitle();
+  useExplorePageTitle(props.queryParams);
```

Comment:
Why do we need to use query params, not the state?

Reply:
good question! To avoid funny behaviors in browser history. When a URL is pushed to the history, the browser also saves the title of the page. if we were to update the URL on state change (therefore slightly before pushing a new item into the history due to how the `useStateSync` hook works), the title would not match the URL. ultimately the URL is the single source of truth from which state is derived, the page title is not different

Reply:
Oh wow, yeah that makes sense! Thanks for explaining it! Would be nice to put a comment in there with that explanation as well (can be in a separate PR)

Reply:
make sense, will do!